mypy/semanal.py

```diff
@@ -1745,6 +1745,8 @@ def analyze_typeddict_classdef(self, defn: ClassDef) -> bool:
             if info is None:
                 self.mark_incomplete(defn.name, defn)
             else:
+                if defn.keywords and "metaclass" in defn.keywords:
```

Comment:
We should also warn for other keywords, as I don't think a TypedDict class can ever take keyword arguments in the class statement. For example, mypy is currently OK with this:

```
from typing import TypedDict

class TD(TypedDict, a=3):
    a: int
```

Reply:
Indeed!